modules/airtable/mentorship_tables.py

```diff
@@ -0,0 +1,185 @@
+            raise valid_e
+
+
+class MentorshipMentorsTable(BaseAirtableTable):
+    def __init__(self):
+        super().__init__("Mentors")
+
+    @cached_property
+    def valid_mentors(self) -> list[Mentor]:
+        """Returns the mentors from the table sorted by row ID
+
+        :return: list of mentors
+        :rtype: list[Mentor]
+        """
+        try:
+            return [self.parse_mentor_row(row) for row in self.all(view="Valid")]
+        except ValidationError as valid_e:
+            raise valid_e
+
+    @staticmethod
+    def parse_mentor_row(row: dict[str, Any]) -> Mentor:
+        fields = {snake_case(k): v for k, v in row["fields"].items()}
+        try:
+            return Mentor(
+                **fields, airtable_id=row["id"], created_at=row["createdTime"]
+            )
+        except ValidationError as valid_e:
+            raise valid_e
+
+
+class MentorshipSkillsetsTable(BaseAirtableTable):
+    def __init__(self):
+        super().__init__("Skillsets")
+
+    @cached_property
+    def valid_skillsets(self) -> list[MentorshipSkillset]:
+        """Returns the skillsets from the table
+
+        :return: list of skillsets
+        :rtype: list[MentorshipSkillset]
+        """
+        try:
+            return [self.parse_skillset_row(row) for row in self.all(view="Valid")]
+        except ValidationError as valid_e:
+            raise valid_e
+
+    @cached_property
+    def mentors_by_skillsets(self) -> dict[str, str]:
+        try:
+            mentors_by_skillset = {}
+            for row in self.all(fields=["Name", "Mentors"], view="Valid"):
+                mentors_by_skillset[row["Name"]] = row["Mentors"]
+            return mentors_by_skillset
+        except Exception as e:
+            logger.warning(f"Issue retrieving mentor IDs by skillset: {e}")
+            raise e
+
+    @staticmethod
+    def parse_skillset_row(row: dict[str, Any]) -> MentorshipSkillset:
+        fields = {snake_case(k): v for k, v in row["fields"].items()}
+        try:
+            return MentorshipSkillset(
+                **fields, airtable_id=row["id"], created_at=row["createdTime"]
+            )
+        except ValidationError as valid_e:
+            raise valid_e
+
+    def mentors_by_skillset(self, skillsets_to_search: list[str]) -> set[str]:
+        logger.info("STAGE: Returning mentors by skillset...")
+        try:
+            mentors = []
+            formula = [f"{{Name}} = '{skillset}'," for skillset in skillsets_to_search]
+            for row in self.all(
+                fields=["Name", "Mentors"],
+                view="Valid",
+                formula=("OR(" + "".join(formula)[:-1] + ")"),
+            ):
+                try:
+                    mentors.append(
+                        row["fields"]["Mentors"] if row["fields"]["Mentors"] else []
+                    )
+                except KeyError as key_e:
+                    logger.warning(f"Key error intercepted: {key_e}")
```

Comment:
I have a feeling no one will ever see this error message 😂  I know we have sentry set up, I wonder if there's a way to alert on warnings and errors?

Reply:
There definitely is. That's also on my list to add, I added it in a couple places but need to finish.